Humberto Raya - HTTP- AJAX

hraya opened the pull request:
@katiagilligan888

hraya committed:
Initial Commit

completed beginning set up for app

hraya committed:
App Renders properly

all steps completed up to the POST

needs to be styled

hraya committed:
Corrected bug

input fields now clear

katiagilligan888 commented on friends/src/App.js:
```diff
@@ -0,0 +1,78 @@
+    event.preventDefault();
+  
+    const Friendz = {
+      id:this.state.friends.length + 1,
```
There is no need to set a value for the ID, since it is created automatically on the server

katiagilligan888 commented on friends/src/App.js:
```diff
@@ -0,0 +1,78 @@
+          {this.state.friends.map(item => (
+              <Friends key={item.id} friends={item} />
+          ))}
+          <form >
```
You could make the form into a separate component

katiagilligan888 commented on friends/src/components/Friends.js:
```diff
@@ -0,0 +1,18 @@
+    render() {
+        return(
+            <div>
+            <p>Name: {this.props.friends.name}</p>
```
Try using destructuring here such as : {name, age, email} = this.props.friends

hraya committed:
Added Delete

App able to delete friends

added delete button for functionality

hraya committed:
Adding update function

added an update button but is not functional yet,

added Form. js have not implemented yet

hraya committed:
Reformatted Form

working on update still

hraya committed:
Completed MVP

Update function is working

Attempting the Stretch Goals

katiagilligan888 commented on friends/src/App.js:
```diff
@@ -0,0 +1,107 @@
+    event.preventDefault();
+    const Friendz = {
+      name:this.state.name,
+      age:this.state.age,
```
Use descructuring here to avoid repeating this.state many times

katiagilligan888 commented on friends/src/App.js:
```diff
@@ -0,0 +1,107 @@
+    updateFriend = (event) => {
+      axios.put(`http://localhost:5000/friends/${event.target.id}`,{
+          name: this.state.name,
+          age:this.state.age,
```
Good job implementing update and delete functionality. Everything looks good!!
